control_plane/src/local_env.rs

```diff
@@ -97,81 +86,158 @@ pub fn init() -> Result<()> {
     let pg_distrib_dir = cargo_path.join("tmp_install");
     let pg_path = pg_distrib_dir.join("bin/postgres");
     if !pg_path.exists() {
-        let err_msg = format!(
-            "Can't find postres binary at {}. \
-            Perhaps './pgbuild.sh' is needed to build it first.",
-            pg_path.to_str().unwrap()
-        );
-        return Err(err_msg.into());
+        anyhow::bail!("Can't find postres binary at {}. \
+                       Perhaps './pgbuild.sh' is needed to build it first.",
+                      pg_path.to_str().unwrap());
     }
 
     // check pageserver
     let zenith_distrib_dir = cargo_path.join("target/debug/");
     let pageserver_path = zenith_distrib_dir.join("pageserver");
     if !pageserver_path.exists() {
-        let err_msg = format!(
-            "Can't find pageserver binary at {}. Please build it.",
-            pageserver_path.to_str().unwrap()
-        );
-        return Err(err_msg.into());
+        anyhow::bail!("Can't find pageserver binary at {}. Please build it.",
+                      pageserver_path.to_str().unwrap());
     }
 
     // ok, we are good to go
-
-    // create dirs
-    let data_dir = cargo_path.join("tmp_check_cli");
-
-    for &dir in &["compute", "pageserver"] {
-        fs::create_dir_all(data_dir.join(dir)).map_err(|e| {
-            format!(
-                "Failed to create directory in '{}': {}",
-                data_dir.to_str().unwrap(),
-                e
-            )
-        })?;
-    }
-
-    // write config
-    let conf = LocalEnv {
-        data_dir,
+    let mut conf = LocalEnv {
+        repo_path: repo_path.clone(),
         pg_distrib_dir,
         zenith_distrib_dir,
+        systemid: 0,
     };
-    let toml = toml::to_string(&conf)?;
-    fs::write(cfg_path, toml)?;
+    init_repo(&mut conf)?;
+
+    Ok(())
+}
+
+pub fn init_repo(local_env: &mut LocalEnv) -> Result<()>
+{
+    let repopath = String::from(local_env.repo_path.to_str().unwrap());
+    fs::create_dir(&repopath).with_context(|| format!("could not create directory {}", repopath))?;
+    fs::create_dir(repopath.clone() + "/pgdatadirs")?;
+    fs::create_dir(repopath.clone() + "/timelines")?;
+    fs::create_dir(repopath.clone() + "/refs")?;
+    fs::create_dir(repopath.clone() + "/refs/branches")?;
+    fs::create_dir(repopath.clone() + "/refs/tags")?;
```

Comment:
Seems like `repopath` should be a `Path` or a `PathBuf` and then `.join("pgdatadirs")` etc.